tests/drivers/rtc/rtc_api/src/test_update_callback.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2022 Bjarki Arge Andreasen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/rtc.h>
+#include <zephyr/sys/atomic.h>
+
+static const struct device *rtc = DEVICE_DT_GET(DT_ALIAS(rtc));
+static atomic_t callback_called_counter;
+static atomic_t callback_test_user_data_address;
+static uint32_t test_user_data = 0x1234;
+
+static void test_rtc_update_callback_handler(const struct device *dev, void *user_data)
+{
+	atomic_inc(&callback_called_counter);
+
+	atomic_set(&callback_test_user_data_address, (uint32_t)user_data);
+}
+
+ZTEST(rtc_api, test_update_callback)
+{
+	int ret;
+	uint32_t counter;
+	uint32_t address;
+
+	ret = rtc_update_set_callback(rtc, NULL, NULL);
+
+	zassert_true(ret == 0, "Failed to clear and disable update callback");
+
+	atomic_set(&callback_called_counter, 0);
+
+	k_msleep(5000);
+
+	counter = atomic_get(&callback_called_counter);
+
+	zassert_true(counter == 0, "Update callback should not have been called");
+
+	ret = rtc_update_set_callback(rtc, test_rtc_update_callback_handler, &test_user_data);
```

Comment:
Doesn't this require that an alarm has been set with `rtc_alarm_set_time()`?  Seems like this test depends on one of the other tests setting the alarm.

Reply:
This is the update callback test, the update event occurs every second synchronized with the 1Hz RTC clock, unlike an alarm callback, which would require a specific alarm to be set :)

Setting the update callback implicitly enables the update event.

Reply:
Thanks for the clarification.

Reply:
Here is a description of the DS12885 **alarm**: "The don’t-care codes in all three alarm bytes create **an
interrupt every second.**"

How is this different than the DS12885's alarm with no bits set in alarm_mask?  It appears the update event is just an alarm with a specific mask set.